torch/ao/sparsity/experimental/data_sparsifier/base_data_sparsifier.py

```diff
@@ -259,3 +253,18 @@ def step(self):
     @abc.abstractmethod
     def update_mask(self, name, data, **kwargs):
         pass
+
+    def __delete_data(self, name):
```

Comment:
nit: why are we using `__` (double underscore?) is this trying to say it's private? I think the convention is single underscore `_`.